### PR TITLE
Ports 12/5/2022

### PR DESCRIPTION
**Pull request by Jamini**
## About The Pull Request

Upstream Merge with some janky adjustments to make it compile.
Expect things that are partially broken or don't function.
Tested, compiles, loads, no runtimes. Some new features are not fully implemented or need to be fixed.
More work to come. Seeking parity and over a thousand PRs to review.

PR's ported

17a13465c Update stealth and camouflage items.dm (#6672)
1ff11c5aa Recoil cooldown change (#6645)
fc804e448 Uplink changes and carrion uplink additions (#6653)
b0d24bf25 check if list is empty before doing pick() (#6656)
7f5b3db2f Remove ".35 Special" (#6661)
cc907b718 Update antagonist.dm (#6652)
6184866ad Properly fixes roaches (#6646)
c58cbfef9 VSC cleanup code (#6622)
a7f738356 Blacklists of most of the hastily added armor disks try 2 (#6664)
d931c954a Prosthetic surgery tweak (#6660)
be5790a1b removed nanakocode (#6655)
446327cab Update airlock_control.dm (#6650)
5c41d69d0 Unlink disposal and trunk when deconstructed (#6649)
b7887ac71 I HATE HUDS THAT WORK BASED OFF BYOND BOGS I HATE HUDS THAT WORK BASED OFF BYOND BUGS (#6641)
52b085b39 Penetration balancing, shield functionality rework (#6613)
ae4853ef0 Fixes/(Buffs..?) Mech-Fabs inheriting materials cost from base tool types, buffs armor, ports a new move sound and chassis type (#6594)
0c202caa1 IH webbed armor; IH turtlenecks (#6640) - Turtlenecks renamed to CAS
49be0f2da Flaregun shotgun (#6591) - FLARES NEED ADDED PROPERLY
a04576f89 Adds bullet resistance to airlocks. Buffs blast doors. (#6630)
1c0aab3dd Put taperoll on any door (#6642)
41b0bb99b tachammer on_suit sprite fix (#6632) - Manually added
1b51005fb Ambience (#6610)
2fa2d0863 Fix job ban pannel (#6581)
311f37a25 Redoes lense code , Thermal lenses no longer provide night vision. (#6607)
add94418c Explosions now knock people back (#6611)
c6b304e10 Hopefully  fixes a mech related runtime + mech buffs (#6572)
e7da2fb5d Shell color and update to M12 sprites (#6620)
55955a950 tacreload fix (#6623)
b460a37ed Meat spike rebalance and readd of crafting (#6601) - MEATSPIKE HAS FUNCTIONALITY ISSUES
c269d5bfb Uplink implant is now hidden (#6552)

## Why It's Good For The Game

Upstream updates 

## Changelog
```changelog
fix: upstream updates
```

### CODE REVIEW

File: code/game/objects/structures/kitchen_spike.dm [hunk 1]
```diff
@@ -19,15 +21,24 @@
...
+	var/list/damaged = H.get_damaged_organs(TRUE, FALSE)
+	for(var/obj/item/organ/external/chest/G in damaged)
+		if(G.brute_dam > 200)
+			to_chat(user, "[H] is too badly damaged to hold onto the meat spike.")
+			return 
+	visible_message(SPAN_DANGER("[user] is trying to force \the [target] onto \the [src]!"))
+	if(do_after(user, 80))
+		if(spike(target))
+			visible_message(SPAN_DANGER("[user] has forced [target] onto \the [src], killing them instantly!"))
+			target.damage_through_armor(201, BRUTE, BP_CHEST)
```
Comment (Shadow-Quill) [inline code/game/objects/structures/kitchen_spike.dm:34]: Eight seconds with a meatspike = instakill, though I doubt anyone here would use this.

**Review by Shadow-Quill — APPROVED** [review]
Didn't see anything too egregious.

File: code/game/objects/structures/kitchen_spike.dm (hunk 1 above)
Comment (Jamini) [reply code/game/objects/structures/kitchen_spike.dm:34]: Hilariously, it doesn't actually work at all.

Comment (Jamini) [reply code/game/objects/structures/kitchen_spike.dm:34]: I'm pretty sure we didn't get all the code for it, or something is broken. At least it didn't work during testing. 

Which is part of why I didn't fool with it (plus there are eight more Prs around Eris' Guru fetish past this)

**Pull request merged**